src/devices/arexx_ml.c

```diff
@@ -103,26 +103,51 @@ static int arexx_ml_decode(r_device *decoder, bitbuffer_t *bitbuffer)
     // Extract data from buffer
     int id       = (b[2] << 8) | (b[1]); // little-endian
     int sens_val = (b[3] << 8) | (b[4]); // big-endian?
-    int is_humi  = id & 1; // even number: Temperature, odd number: Humidity
 
-    // MCP9808 Ambient Temperature Register "5-4":
-    int temp_alert = (sens_val >> 13) & 7;
-    int temp_raw   = (int16_t)(sens_val << 3); // uses sign-extend
-    float temp_c   = temp_raw / 128;
+    // Decode readings
+    float temp_c = 0.0, humidity = 0.0;
+    int is_humi = 0, is_temp = 0, is_alert = 0, temp_alert = 0;
+
+    switch (msg_len) {
+    case 5:
+        if ((id & 0xF000) == 0x2000) {
```

Comment:
Make this more plain, e.g. each check like `if (msg_len == 5 && (id & 0xF000) == 0x2000) {`